engine/time/src/main/java/io/deephaven/time/DateTimeUtils.java

```diff
@@ -2336,4 +2376,38 @@ public static String getPartitionFromTimestampSeconds(@NotNull final DateTimeFor
         }
         return dateTimeFormatter.format(Instant.ofEpochMilli(timestampSeconds * 1_000));
     }
+
+    private enum TimeProviderImpl implements TimeProvider {
```

Comment:
why enum?

I think it'd be more straightforward to just add this INSTANCE as a field in TimeProvider

Reply:
IMO, when singleton structures can be represented as enum, you gain niceties - equality, hashcode, etc for free.